packages/core/echo/echo-db/src/proxy-db/mutable-schema-registry.ts

```diff
@@ -166,9 +166,21 @@ export class MutableSchemaRegistry implements SchemaResolver {
       return existing;
     }
 
+    let previousTypename: string | undefined;
+
     const mutableSchema = new MutableSchema(schema);
     const subscription = getObjectCore(schema).updates.on(() => {
       mutableSchema.invalidate();
+
+      if (previousTypename != undefined && schema.typename !== previousTypename) {
```

Comment:
I think != null? (i don't love the != syntax, but I've only seen it used with null -- to mean both null and undefined)
can we avoid the check here with just schema.typename !== previousTypename?